### PR TITLE
KP add proxy tutorial

### PR DESCRIPTION
**Pull request by intelkevinputnam**
In progress.

To do:
    * Redsocks
    * Transitioning between proxy and non-proxy environments
    * Shells and programs in a desktop session: Gnome desktop
    * Firefox

### CODE REVIEW

File: source/clear-linux/tutorials/tutorial-proxy.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+Terminal
+========
+
+Add the following to your ~/.bashrc:
+
+.. code-block:: bash
+
+    export http_proxy=http://your.http-proxy.url:port
+    export https_proxy=http://your.https-proxy.url:port
+    export no_proxy=".your-company-domain.com,localhost"
```
Comment (tpleavitt) [inline source/clear-linux/tutorials/tutorial-proxy.rst]: Also need to include the capitalized versions of the env variables:
export HTTP_PROXY=$http_proxy
same for HTTPS_PROXY and NO_PROXY.  You can also add variables for fttp_proxy/FTP_PROXY and socks_proxy/SOCKS_PROXY

Comment (intelkevinputnam) [reply source/clear-linux/tutorials/tutorial-proxy.rst]: This cover it?

export http_proxy=http://your.http-proxy.url:port
export https_proxy=http://your.https-proxy.url:port
export ftp_proxy=http://your.ftp-proxy.url:port
export socks_proxy=http://your.socks-proxy.url:port
export no_proxy=".your-company-domain.com,localhost"
export HTTP_PROXY=$http_proxy
export HTTPS_PROXY=$https_proxy
export FTP_PROXY=$ftp_proxy
export SOCKS_PROXY=$socks_proxy
export NO_PROXY=$no_proxy

Comment (tpleavitt) [reply source/clear-linux/tutorials/tutorial-proxy.rst]: Yep, that's how I do it

File: source/clear-linux/tutorials/tutorial-proxy.rst [hunk 2]
```diff
@@ -0,0 +1,103 @@
+.. _tutorial-proxy:
+
+Setting up proxy
+################
+
+This tutorial shows you how to configure your system for use behind a
+corporate proxy.
```
Comment (puneetse) [inline source/clear-linux/tutorials/tutorial-proxy.rst]:
```suggestion
outbound network proxy to access the Internet. 
```

It's not just corporations that use proxies.

File: source/clear-linux/tutorials/tutorial-proxy.rst [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    http_proxy = your.http-proxy.url:port
+    https_proxy = your.https-proxy.url:port
+    ftp_proxy = your.http-proxy.url:port
+    no_proxy = your-company-domain.com, localhost
+    EOF
+
+System service (Docker)
+***********************
+
+For Docker (and other services that use systemd), you can follow these steps to configure and check proxy settings:
```
Comment (puneetse) [inline source/clear-linux/tutorials/tutorial-proxy.rst]: I suggest clarifying or adding a note here that this proxy is for the Docker service network connections (e.g. `docker pull`). Networking inside of a deployed container has different mechanisms for defining proxy information, depending on the image. (See: https://docs.docker.com/network/proxy/)

File: source/clear-linux/tutorials/tutorial-proxy.rst [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    .. code-block:: bash
+
+        sudo systemctl daemon-reload
+        sudo systemctl restart docker
+
+#. Verify that changes have loaded:
+
+    .. code-block:: bash
+
+        systemctl show --property=Environment docker
```
Comment (puneetse) [inline source/clear-linux/tutorials/tutorial-proxy.rst]: This is cool! 👍

File: source/clear-linux/tutorials/tutorial-proxy.rst [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        Environment=HTTP_PROXY=http://your.http-proxy.url:port/ HTTPS_PROXY=http://your.https-proxy.url:port/
+
+git over ssh
+************
+
+Add the following to your :file:`~/.ssh/config`:
+
+.. code-block:: bash
+
+    host github.com
```
Comment (puneetse) [inline source/clear-linux/tutorials/tutorial-proxy.rst]: The syntax formatting should have everything under `host` indented/spaced as a subsection.

File: source/clear-linux/tutorials/tutorial-proxy.rst [hunk 6]
```diff
@@ -100,9 +105,9 @@ Add the following to your :file:`~/.ssh/config`:
...
 
     host github.com
-    port 22    
-    user git
-    ProxyCommand connect-proxy -S your.ssh-proxy.url:port %h %p
+        port 22    
+        user git
+        ProxyCommand connect-proxy -S your.ssh-proxy.url:port %h %p
 
 .. note::
```
Comment (tpleavitt) [inline source/clear-linux/tutorials/tutorial-proxy.rst]: In the note.  Change thought to though

Comment (intelkevinputnam) [reply source/clear-linux/tutorials/tutorial-proxy.rst]: good catch. Fixed.

**Comment by intelkevinputnam** [comment]
@puneetse Please take a look and review updates.

**Pull request merged**